### PR TITLE
feat(plugins/adr): support for i18n feature

### PR DESCRIPTION
**Pull request by axdotl**
## Hey, I just made a Pull Request!

This change introduces the support for i18n feature for the ADR plugin.

As a side effect the content header for the `EntityAdrContent` becomes configurable.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by aksdb** [comment]
Even though I don't have a vote here, I would just like to leave my two cents: having an even better solution in the future is nice, but from what I can tell this PR is simple enough to not have any ugly side-effects (the implementation is backwards compatible and quite simple). It doesn't prevent a better solution, but offers a relief in the meantime. Basically "perfect is the enemy of good". This solution might not be perfect, but IMO it's already good enough and therefore takes away pressure from implementing a larger, more wide-reaching change, without preventing such a change either.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you @kuangp, indeed.

Good news though, #17436  is now merged 😁 @axdotl do you want to implement this using a translation ref instead?

**Comment by axdotl** [comment]
@Rugvip Sure, I'll give it a try and will update the PR.

**Comment by axdotl** [comment]
@Rugvip I rebased and updated the PR.

File: plugins/adr/src/components/EntityAdrContent/EntityAdrContent.tsx [hunk 1]
```diff
@@ -214,7 +217,7 @@ export const EntityAdrContent = (props: {
 
   return (
     <Content>
-      <ContentHeader title="Architecture Decision Records">
+      <ContentHeader title={t('content_header_title')}>
```
Comment (kuangp) [inline plugins/adr/src/components/EntityAdrContent/EntityAdrContent.tsx:220]: hm I'm not a big fan of using free form strings as "keys" for translation messages - I'm wondering if we could establish some kind of pattern where this has better type checks eg. through restricting the type of this param to a set list of enums based off keys of the messages in the translation ref 🤔 

Not something that needs to be addressed in this PR but just thinking out loud and would be curious on your thoughts @Rugvip @mario-mui

File: plugins/adr/src/translations.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createTranslationRef } from '@backstage/core-plugin-api/alpha';
+
+/** @alpha */
+export const adrTranslationRef = createTranslationRef({
```
Comment (kuangp) [inline plugins/adr/src/translations.ts:19]: I think this needs to be exported from this package so consumers can modify the translations? It looks like the current convention is to ship it via [`alpha` exports](https://github.com/backstage/backstage/pull/17436/files#diff-6545ef79075af1d48222e636187d3bf1e24e825926c0e98357300ed8dc22c6ab) as this is still experimental

**Review by kuangp — CHANGES_REQUESTED** [review]

File: plugins/adr/src/components/EntityAdrContent/EntityAdrContent.tsx (hunk 1 above)
Comment (axdotl) [reply plugins/adr/src/components/EntityAdrContent/EntityAdrContent.tsx:220]: Well, I'm open to suggestions. I just tried to apply the pattern from the UserSettings example.

File: plugins/adr/src/translations.ts (hunk 2 above)
Comment (axdotl) [reply plugins/adr/src/translations.ts:19]: Do I miss something? I thought line 18 marks the export as `alpha`, isn't it?

Comment (kuangp) [reply plugins/adr/src/translations.ts:19]: Yes - marking it `@alpha` generates an alpha export for the `api-report` (which reminds me, you will also need to update the `api-reports.md` by running `yarn build:api-reports` from the repo root, after you make the changes below) 

I meant that you need to actually export this symbol from the package itself - for example see the change [here](https://github.com/backstage/backstage/pull/17436/files#diff-6545ef79075af1d48222e636187d3bf1e24e825926c0e98357300ed8dc22c6ab) which exports things from an `alpha.ts` file and also the associated [changes here](https://github.com/backstage/backstage/pull/17436/files#diff-e1b268e9d36dc2d5ad582a461567cd08f5807b61457a282a414490323c77ad5fR8-R24) you would also need to make to export these alpha symbols from this package

File: plugins/adr/src/components/EntityAdrContent/EntityAdrContent.tsx (hunk 1 above)
Comment (mario-mui) [reply plugins/adr/src/components/EntityAdrContent/EntityAdrContent.tsx:220]: > hm I'm not a big fan of using free form strings as "keys" for translation messages - I'm wondering if we could establish some kind of pattern where this has better type checks eg. through restricting the type of this param to a set list of enums based off keys of the messages in the translation ref 🤔
> 
> Not something that needs to be addressed in this PR but just thinking out loud and would be curious on your thoughts @Rugvip @mario-mui

image
image
As I thought, we can get key's prompts by writing comment.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍

File: plugins/adr/src/translations.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  id: 'adr',
+  messages: {
+    content_header_title: 'Architecture Decision Records',
+    failed_to_fetch: 'Failed to fetch ADRs',
+    no_adrs: 'No ADRs found',
+  },
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply plugins/adr/src/translations.ts:25]: Ah sorry I'm confused, this is about getting type hints for the default message, not the key set. Still best to add 😁

**Comment by axdotl** [comment]
Thanks a lot for your feedback and review - I hope I have addressed all your remarks.

**Review by Rugvip — APPROVED** [review]
Looks good, thank you! 👍

**Comment by axdotl** [comment]
Thanks for your support 💙

**Pull request merged**